api/apps/v1alpha1/nimbuild_types.go

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// NIMBuildSpec to build optimized trtllm engines with given model config and weights.
+type NIMBuildSpec struct {
+	// Required: Reference to the model weights from NIMCache
+	NIMCacheName string `json:"nimCacheName"`
+	// Profile is the name of the profile to be used for building the engine.
+	Profile string `json:"profile,omitempty"`
+	// EngineName is the name given to the engine being built.
+	EngineName string `json:"engineName,omitempty"`
+	// Resources is the resource requirements for the NIMBuild pod.
+	Resources *corev1.ResourceRequirements `json:"resources,omitempty"`
+	// Tolerations for running the job to cache the NIM model
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+	// NodeSelector is the node selector labels to schedule the caching job.
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+}
+
+// NIMBuildStatus defines the observed state of NIMBuild.
+type NIMBuildStatus struct {
+	State         string             `json:"state,omitempty"`
+	TargetProfile NIMProfile         `json:"targetProfile,omitempty"`
+	BuiltProfile  NIMProfile         `json:"builtProfile,omitempty"`
+	Conditions    []metav1.Condition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type" protobuf:"bytes,1,rep,name=conditions"`
+}
+
+// +genclient
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Status",type=string,JSONPath=`.status.state`,priority=0
+// +kubebuilder:printcolumn:name="Age",type="date",format="date-time",JSONPath=".metadata.creationTimestamp",priority=0
+
+// NIMBuild is the Schema for the nimcaches API.
+type NIMBuild struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NIMBuildSpec   `json:"spec,omitempty"`
```

Comment:
```suggestion
       // +kubebuilder:validation:XValidation:rule="self == oldSelf", message="spec is immutable"
	Spec   NIMBuildSpec   `json:"spec,omitempty"`
```
Immutability should be handled here.